src/ngraph/builder/reshape.cpp

```diff
@@ -258,58 +258,37 @@ namespace ngraph
 
 shared_ptr<Node> builder::opset1::flatten(const Output<Node>& value, int axis)
 {
-    if (value.get_partial_shape().is_static())
+    // First dimension of output tensor is the product of [d_0, ... d_{axis-1}] dimensions of
+    // input tensor. The last dimension is the product of the rest of input tensor dimensions:
+    // [d_{axis}, ..., d_n]
+    shared_ptr<Node> output_shape;
+    if (axis == 0)
     {
-        auto data_shape = value.get_shape();
-        // First dimension of output tensor is the product of [d_0, ... d_{axis-1}] dimensions of
-        // input
-        // tensor. The last dimension is the product of the rest of input tensor dimensions:
-        // [d_{axis}, ..., d_n]
-        size_t first_dim_size =
-            accumulate(begin(data_shape), next(begin(data_shape), axis), 1UL, multiplies<size_t>());
-
-        size_t last_dim_size =
-            accumulate(next(begin(data_shape), axis), end(data_shape), 1UL, multiplies<size_t>());
-
-        return builder::opset1::reshape(value, Shape{first_dim_size, last_dim_size});
+        output_shape = ngraph::opset1::Constant::create(element::i64, Shape{2}, {1, -1});
     }
     else
     {
-        shared_ptr<Node> output_shape;
-        if (axis == 0)
-        {
-            output_shape = ngraph::opset1::Constant::create(element::i64, Shape{2}, {1, -1});
-        }
-        else if (axis == 1)
```

Comment:
This branch should stay. Please revert it back.

Reply:
done